File: system-http-client/src/main/java/com/levnovikov/core_network/request/RequestBody.java
package com.levnovikov.core_network.request;

/**
 * Author: lev.novikov
 * Date: 27/3/18.
 */

public class RequestBody {

    private String content = "";
    private String contentType = "";

    public String getContent() {
        return content;
    }

    public RequestBody() {}

    public RequestBody(String content, String contentType) {
        this.content = content;
        this.contentType = contentType;
    }

    public void setContent(String content, String contentType) {
        this.content = content;
        this.contentType = contentType;
    }

    public boolean isNotEmpty() {
        return !content.isEmpty();
    }

    public RequestBody copy() {
        RequestBody body = new RequestBody();
        body.content = content;
        body.contentType = contentType;
        return body;
    }
}

File: system-http-client/src/main/java/com/levnovikov/core_network/request/RequestBody.kt
package com.levnovikov.core_network.request

/**
 * Author: lev.novikov
 * Date: 27/3/18.
 */

class RequestBody {

    var content = ""
        private set
    private var contentType = ""

    val isNotEmpty: Boolean
        get() = !content.isEmpty()

    constructor()

    constructor(content: String, contentType: String) {
        this.content = content
        this.contentType = contentType
    }

    fun setContent(content: String, contentType: String) {
        this.content = content
        this.contentType = contentType
    }

    fun copy(): RequestBody {
        val body = RequestBody()
        body.content = content
        body.contentType = contentType
        return body
    }
}
